rust/protocol/src/sealed_sender.rs

```diff
@@ -1117,6 +1482,15 @@ pub async fn sealed_sender_decrypt_to_usmc(
             authentication_tag,
             encrypted_message,
         } => {
+            let encrypted_message_key: [u8; 32] = encrypted_message_key
+                .as_ref()
+                .try_into()
+                .map_err(|_: TryFromSliceError| {
+                    SignalProtocolError::BadKeyLength(
+                        curve::KeyType::Djb,
+                        encrypted_message_key.len(),
```

Comment:
Hmm. The message key isn't a curve key. Let's just go with InvalidSealedSenderMessage for now; the errors need attention separately anyway, and in practice this should never happen.

Reply:
That makes sense! I've done this with a `format!(...)` string which prints the detected size of that slice although you've mentioned it should never happen.